python/lsst/ts/standardscripts/maintel/apply_dof.py

```diff
@@ -0,0 +1,430 @@
+                    Units in um.
+                  default: 0.0
+              M2_B14:
+                  type: number
+                  description: >-
+                    Defines the offset applied to the M2 bending mode 14.
+                    Units in um.
+                  default: 0.0
+              M2_B15:
+                  type: number
+                  description: >-
+                    Defines the offset applied to the M2 bending mode 15.
+                    Units in um.
+                  default: 0.0
+              M2_B16:
+                  type: number
+                  description: >-
+                    Defines the offset applied to the M2 bending mode 16.
+                    Units in um.
+                  default: 0.0
+              M2_B17:
+                  type: number
+                  description: >-
+                    Defines the offset applied to the M2 bending mode 17.
+                    Units in um.
+                  default: 0.0
+              M2_B18:
+                  type: number
+                  description: >-
+                    Defines the offset applied to the M2 bending mode 18.
+                    Units in um.
+                  default: 0.0
+              M2_B19:
+                  type: number
+                  description: >-
+                    Defines the offset applied to the M2 bending mode 19.
+                    Units in um.
+                  default: 0.0
+              M2_B20:
+                  type: number
+                  description: >-
+                    Defines the offset applied to the M2 bending mode 20.
+                    Units in um.
+                  default: 0.0
+            additionalProperties: false
+        """
+        return yaml.safe_load(schema_yaml)
+
+    async def configure(self, config) -> None:
+        """Configure script.
+
+        Parameters
+        ----------
+        config : `types.SimpleNamespace`
+            Script configuration, as defined by `schema`.
+        """
+
+        # Configure tcs and camera
+        await self.configure_tcs()
+
+        # Loop through properties and assign their values to the vector
+        for key, value in vars(config).items():
+            self.dofs[getattr(DOFName, key)] = value
+
+    def set_metadata(self, metadata) -> None:
+        """Set script metadata.
+
+        Parameters
+        ----------
+        metadata : `lsst.ts.salobj.base.ScriptMetadata`
+            Script metadata.
+        """
+        metadata.duration = 10
+
+    async def assert_feasibility(self) -> None:
+        """Verify that the telescope is in a feasible state to
+        execute the script.
+        """
+
+        await self.mtcs.assert_all_enabled()
+
+    async def run(self) -> None:
+        """Run script."""
+        # Assert feasibility
+        await self.assert_feasibility()
+
+        await self.checkpoint("Applying DOF offset...")
+        await self.mtcs.mtaos.cmd_offsetDOF(self.dofs)
```

Comment:
This is not going to work. To execute the command you need to call `set_start`, however with arrays like this you will need to do something slightly different. something like this should work:

```
offset_dof_data = self.mtcs.mtaos.cmd_offsetDOF.DataType()
for i, dof_offset in enumerate(self.dofs):
    offset_dof_data.value[i] = dof_offset
await self.mtcs.mtaos.cmd_offsetDOF.start(data=offset_dof_data)
```